experiments/experiment_prototype.py

```diff
@@ -1,385 +1,1111 @@
+            assert isinstance(exp_slice['mpinc'], int)
+        except AssertionError:
+            errmsg = "Slice must specify mpinc that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:  # TODO may want to add a field for range_gate which could set this param.
+            assert 'pulse_len' in exp_slice.keys()
+            assert isinstance(exp_slice['pulse_len'], int)
+        except AssertionError:
+            errmsg = "Slice must specify pulse_len that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'nrang' in exp_slice.keys()
+            assert isinstance(exp_slice['nrang'], int)
+        except AssertionError:
+            errmsg = "Slice must specify nrang that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'frang' in exp_slice.keys()
+            assert isinstance(exp_slice['frang'], int)
+        except AssertionError:
+            errmsg = "Slice must specify frang that must be an integer"
+            raise ExperimentException(errmsg, exp_slice)
+
+        try:
+            assert 'intt' in exp_slice.keys()
+        except AssertionError:
+            try:
+                assert 'intn' in exp_slice.keys()
+            except AssertionError:
+                errmsg = "Slice must specify either an intn or intt"
+                raise ExperimentException(errmsg, exp_slice)
+            else:
+                try:
+                    assert isinstance(exp_slice['intn'], int)
+                except AssertionError:
+                    errmsg = "intn must be an integer"
+                    raise ExperimentException(errmsg, exp_slice)
+        else:
+            try:
+                assert isinstance(exp_slice['intt'], int)
+            except AssertionError:
+                errmsg = "intt must be an integer"
+                raise ExperimentException(errmsg, exp_slice)
+            else:
+                try:
+                    assert 'intn' in exp_slice.keys()
+                except AssertionError:
+                    pass
+                else:
+                    # TODO Log warning intn will not be used
+                    exp_slice.pop('intn')
+
+        try:
+            assert 'beam_angle' in exp_slice.keys()
+            assert isinstance(exp_slice['beam_angle'], list)
+            for element in exp_slice['beam_angle']:
+                assert isinstance(element, float) or isinstance(element, int) or isinstance(element,
+                                                                                            list)
+                if isinstance(element, list):
+                    for angle in element:
+                        assert isinstance(angle, float) or isinstance(angle, int)
+        except AssertionError:
+            errmsg = "Slice must specify beam_angle that must be a list of floats or lists (of floats)"
```

Comment:
errmsg doesn't match asserts, you're also checking if they are ints, not just floats.